### PR TITLE
Handle sub/super-scripts in rich text

### LINKED ISSUE #14317: Text size in annotations isn't supoorted correctly
Attach (recommended) or Link to PDF file here: 

[anootation_problem.pdf](https://github.com/mozilla/pdf.js/files/7611701/anootation_problem.pdf)

Configuration:
- Web browser and its version: 98.0.4732.0 (Official Build) canary (64-bit)
- Operating system and its version: Edition Windows 10 Home, Version 20H2, OS build 19042.1348
- PDF.js version: 2.12.211
- Is a browser extension: Yes

Steps to reproduce the problem:
1. Open pdf file
2. Scroll to bottom of page and point to annotation of 'loss ft(x) >= 0,' text
3. Compare the displayed text with same in Acrobat Reader

What is the expected behavior? (add screenshot)
image

What went wrong? (add screenshot)
Text size not respected (e.g., in superscripts and subscripts)
image

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by calixteman**
- it aims to fix #14317;
 - change the fontSize and the verticalAlign properties according to the position of the text.

### CODE REVIEW

File: test/test_manifest.json [hunk 1]
```diff
@@ -159,6 +159,14 @@
...
        "annotations": true
     },
+    {  "id": "issue14317",
+      "file": "pdfs/issue14317.pdf",
+      "md5": "581c807aaf604542b2dafc1f814dc6c8",
+      "rounds": 1,
+      "lastPage": 1,
+      "type": "eq",
+      "annotations": true
+   },
```
Comment (Snuffleupagus) [inline test/test_manifest.json:169]: Is this really necessary, since I believe that `issue13915` should cover this already?

**Comment by calixteman** [comment]
The patch fixes an issue in xfa too: `vertical-align` is allowed in the css properties but it was filtered out.
So the "regressions" are finally expected here.

**Pull request merged**